openshift/pipeline-latest-release.sh

```diff
@@ -14,10 +14,9 @@ function tryurl {
     curl -s -o /dev/null -f ${1} || return 1
 }
 
-if [[ "${1:-}" != "--only-stable-release" ]];then
-  if tryurl ${NIGHTLY_RELEASE};then
+# check for pipeline nightly release if no nightly the go for version based yaml file.
+if tryurl ${NIGHTLY_RELEASE};then
     export RELEASE_YAML=${NIGHTLY_RELEASE}
-  fi
 else
 for shifted in `seq 0 ${MAX_SHIFT}`;do
     versionyaml=$(get_version ${shifted})
```

Comment:
There is a logical bug in this part, I think we should discuss it and then move forward.
Like @khrm pointed out, the `else` statement will pretty much never get executed, it will only get executed when the nightly URL is not reachable for some reason like network issues, etc.
But even if we want to be extra paranoid about having a fallback, there is a logical bug in how the `else` statement will execute.

Like @savitaashture and I discussed yesterday, in the `get_version` function:
https://github.com/openshift/tektoncd-triggers/blob/480cb385c450af7d1cfd30ab7ccb37f0e20a77e4/openshift/pipeline-latest-release.sh#L7-L11
first, it tries to get the releases from upstream `tektoncd/pipeline` - https://github.com/openshift/tektoncd-triggers/blob/480cb385c450af7d1cfd30ab7ccb37f0e20a77e4/openshift/pipeline-latest-release.sh#L9
which results in output like `(v0.27.3, v0.27.2, v0.27.1, ...)` and then it tries to substitute the versions in the line https://github.com/openshift/tektoncd-triggers/blob/480cb385c450af7d1cfd30ab7ccb37f0e20a77e4/openshift/pipeline-latest-release.sh#L4 which are the releases for `openshift/tektoncd-pipeline` which *do not* have those versions available. The latest version at `openshift/tektoncd-pipeline` is `v0.24.3`.

Due to this mismatch of versions between upstream and midstream, the `else` statement will *always* fail - and this is exactly why CI is failing right now and we are getting errors about `RELEASE_YAML` not being set.

We need to either ditch getting versions from upstream and rely solely on the latest midstream version or start releasing midstream more often to avoid this issue.